### ISSUE TITLE
Items deselected when scrolling over a nested ListBox

### ISSUE THREAD

**Issue by TomEdwardsEnscape** [AvaloniaUI/Avalonia#14814]
Add this XAML to a window:

``` xaml
<ListBox Name="Box">
  <ListBox.ItemTemplate>
    <DataTemplate>
      <ListBox ItemsSource="{Binding}"/>
    </DataTemplate>
  </ListBox.ItemTemplate>
</ListBox>
```

And then execute this in code-behind:

```C#
Box.ItemsSource = new IEnumerable<int>[] { Enumerable.Range(0, 50), Enumerable.Range(50, 50) };
```
Then:

1. Select an item near the top of the combined list
2. Scroll to the bottom of the list
3. Scroll back to the top of the list
4. The item you selected will have been deselected

I have also noticed some visual flickering while scrolling, which is likely to be a separate issue.

This issue can be worked around by changing the `ItemsPanel` of all `ListBox` controls to one containing a non-virtualising `StackPanel`.

## Expected behavior
Item selection should not change when scrolling through a virtualised list.

## Environment 

- OS: Windows
- Avalonia-Version: Git master

**Comment by timunie** [comment]
Does it also happen on 11.0? If so, this is a regression imo. 

Probably same as these two:
- https://github.com/AvaloniaUI/Avalonia/discussions/14342
- https://github.com/AvaloniaUI/Avalonia/discussions/14763

**Comment by TomEdwardsEnscape** [comment]
This also happens on 11.0.

**Comment by MrJul** [comment]
I'm not really sure we can do anything here.

When the inner `ListBox` gets recycled, all its properties are lost, and that's expected from the `VirtualizingStackPanel`.
If the inner `SelectedItem` is bound, it's restored correctly when scrolling, and that's what matters imo.

The best we can do is probably to document this as a unsupported edge case.

**Comment by timunie** [comment]
I agree with @MrJul here. Will convert it into a discussion for now.
